packages/next/client/components/reducer.ts

```
@@ -441,101 +506,132 @@ export function reducer(
+            // Don't apply scroll and focus management.
+            focusAndScrollRef: { apply: false },
             cache: state.cache,
             tree: state.tree,
           }
         }
 
+        // Remove cache.data as it has been resolved at this point.
         cache.data = null
 
+        // TODO-APP: Currently the Flight data can only have one item but in the future it can have multiple paths.
         const flightDataPath = flightData[0]
 
+        // The one before last item is the router state tree patch
         const [treePatch] = flightDataPath.slice(-2)
-        const treePath = flightDataPath.slice(0, -3)
+
+        // Path without the last segment, router state, and the subTreeData
+        const flightSegmentPath = flightDataPath.slice(0, -3)
+
+        // Create new tree based on the flightSegmentPath and router state patch
         const newTree = walkTreeWithFlightDataPath(
           // TODO-APP: remove ''
-          ['', ...treePath],
+          ['', ...flightSegmentPath],
           state.tree,
           treePatch
         )
 
         mutable.previousTree = state.tree
         mutable.patchedTree = newTree
 
+        // Copy subTreeData for the root node of the cache.
         cache.subTreeData = state.cache.subTreeData
+        // Create a copy of the existing cache with the subTreeData applied.
         fillCacheWithNewSubTreeData(cache, state.cache, flightDataPath)
 
         return {
+          // Set href.
           canonicalUrl: href,
+          // Set pendingPush.
           pushRef: { pendingPush, mpaNavigation: false },
-          focusRef: { focus: true },
+          // All navigation requires scroll and focus management to trigger.
+          focusAndScrollRef: { apply: true },
+          // Apply patched cache.
           cache: cache,
+          // Apply patched tree.
           tree: newTree,
         }
       }
 
+      // This case should never be hit as `cacheType` is required and both cases are implemented.
```

Should we throw an error here instead given that?